Allow arbitrary (defined) params through

brandonfarber opened the pull request:
We keep running into issues where we need to pass additional parameters through to Adyen, and we have to update this package to accomplish it. While I appreciate its whitelist approach, this has proven restrictive.

This PR allows us to define parameters we want to allow through first (sticking with the "whitelist" approach), and then pass those arbitrary parameters through in the various requests.

Additionally, I added the ability to log the request and response to any arbitrary log channel (i.e. the Laravel log channel) for debug purposes.

brandonfarber committed:
Allow arbitrary (defined) params through

brandonfarber committed:
Introduce ability to log request/response

brandonfarber committed:
Fix bug

brandonfarber committed:
Another bug fix attempt

aalba-vshred committed:
feat(AbstractRequest.php): update payment API versions to v71 for checkout and payment
feat(CreateSessionRequest.php): refactor logic to apply storage parameters for payment method
feat(AbstractRequestTest.php): update test URLs to use dynamic version constants for payments

aalba-vshred committed:
fix(CreateSessionRequest.php): set default value for 'value' key in amount array to getAmountInteger() or 0 if null

aalba-vshred committed:
Merge branch 'feature/v71-upgrade' into fix/OTT-1714-paypal

aalba-vshred commented on src/Message/Checkout/CreateSessionRequest.php:
```diff
@@ -32,7 +32,7 @@ public function getData()
         $data = [
             'amount' => [
-                'value' => 0,
+                'value' => $this->getAmountInteger() ?? 0,
```
Calling out updates to @brandonfarber work. This allows us to pass value into create session request.

aalba-vshred commented on src/Message/Checkout/CreateSessionRequest.php:
```diff
@@ -41,9 +41,7 @@ public function getData()
-        if (!empty($this->getStorePaymentMethod())) {
-            $data['storePaymentMethod'] = $this->getStorePaymentMethod();
-        }
+        $data = $this->applyStorageParameters($data);
```
Calling out updates to @brandonfarber work. Added new `applyStorageParamters($data)` to allow us to generate the proper `storePaymentMethod` key/value depending on what is passed (`storePaymentMethodMode` was added after v69)

aalba-vshred commented on src/Message/Checkout/CreateSessionRequest.php:
```diff
@@ -68,4 +69,29 @@ public function getStorePaymentMethod()
         return $this->getParameter('storePaymentMethod');
     }
+
+    public function setStorePaymentMethodMode($storePaymentMethodMode)
```
Calling out updates to @brandonfarber work. Called out above.

aalba-vshred commented on src/Message/AbstractRequest.php:
```diff
@@ -21,9 +54,9 @@ abstract class AbstractRequest extends OmnipayAbstractRequest
 
     const VERSION_DIRECTORY         = 'v2';
-    const VERSION_CHECKOUT          = 'v69';
+    const VERSION_CHECKOUT          = 'v71';
```
Calling out updates to @brandonfarber work. Changed the version here.

aalba-vshred commented on tests/Message/AbstractRequestTest.php:
```diff
@@ -31,7 +31,7 @@ public function testGetCheckoutUrlTestMode()
         
         $this->assertEquals(
-            'https://checkout-test.adyen.com/v69/payments',
+            'https://checkout-test.adyen.com/' . AbstractRequest::VERSION_CHECKOUT . '/payments',
```
Calling out updates to @brandonfarber work. Updated this to pull in version so it does not get missed for testing.

aalba-vshred reviewed (approved):
I have approved this, but hoping @brandonfarber will take another look to ensure this is correct.

aalba-vshred committed:
refactor(AuthorizeRequest.php): refactor getPaymentMethodData and getCardData methods to return type array
feat(AuthorizeRequest.php): add support for setting and getting capture delay hours with type hinting
feat(AuthorizeRequest.php): add method getRequestedTestAcquirerResponseCode to retrieve test acquirer response code

aalba-vshred committed:
Merge branch 'feature/v71-upgrade' into fix/OTT-1714-paypal
